smdebug/core/config_validator.py

```diff
@@ -0,0 +1,106 @@
+# Standard Library
+import os
+from distutils.util import strtobool
+from enum import Enum
+
+# First Party
+import smdebug.core.utils
+from smdebug.core.logger import get_logger
+from smdebug.core.utils import is_framework_version_supported
+from smdebug.profiler.profiler_config_parser import ProfilerConfigParser
+
+logger = get_logger()
+
+_config_validator = None
+
+
+class FRAMEWORK(Enum):
+    PYTORCH = 0
+    TENSORFLOW = 1
+    MXNET = 2
+    XGBOOST = 3
+
+
+SupportedFrameworks = [FRAMEWORK.PYTORCH, FRAMEWORK.TENSORFLOW, FRAMEWORK.MXNET, FRAMEWORK.XGBOOST]
+
+
+class ConfigValidator(object):
+    def __init__(self, frameworkType: FRAMEWORK):
+        self._create_hook = strtobool(os.getenv("USE_SMDEBUG", "true").lower())
+        self._summary = ""
+        self._frameworkType = frameworkType
+
+    def _validate_training_environment(self):
+        """
+        The function checks whether the debugger functionality can be supported in the given training environment.
+        Currently, if the training job is using a framework that is not supported by the debugger, we disable the
+        creation of hook and disable the debugger functionality.
+        :return:
+        """
+        if (
+            self._frameworkType in SupportedFrameworks
```

Comment:
We don't need this check anymore. By enforcing that the argument is of type `FRAMEWORK` we already guarantee that it is a supported framework.

Reply:
This is needed because we have to ensure that the specific version of framework is supported or not.